src/go/rpk/pkg/cli/cluster/connections/list.go

```diff
@@ -0,0 +1,210 @@
+	closed := conn.CloseTime.AsTime()
+
+	duration := closed.Sub(opened)
+	if opened.After(closed) {
+		duration = time.Since(opened)
+	}
+
+	return duration.Round(time.Second).String()
+}
+
+func writeConnectionRow(w *out.TabWriter, conn *adminv2.KafkaConnection) {
+	w.Print(
+		conn.NodeId,
+		conn.ShardId,
+		conn.Uid,
+		conn.ClientId,
+		fmt.Sprintf("%s:%d", conn.Source.IpAddress, conn.Source.Port),
+		getConnectionDuration(conn),
+		conn.IdleDuration.AsDuration().Round(time.Millisecond).String(),
+		units.HumanSize(float64(conn.RecentRequestStatistics.ProduceBytes/60)),
+		units.HumanSize(float64(conn.RecentRequestStatistics.FetchBytes/60)),
+		conn.RecentRequestStatistics.RequestCount,
+		strings.Replace(conn.State.String(), "KAFKA_CONNECTION_STATE_", "", 1),
+	)
+}
+
+func printConnectionListTable(resp *adminv2.ListKafkaConnectionsResponse) string {
+	if resp.TotalSize < 1 || resp.Connections == nil || len(resp.Connections) == 0 {
+		return "No open connections found."
+	}
+
+	var buf bytes.Buffer
+	writer := out.NewTableTo(&buf, tableHeaders...)
+
+	for _, conn := range resp.Connections {
+		writeConnectionRow(writer, conn)
+	}
+
+	writer.Flush()
+	return buf.String()
+}
+
+func buildFilters(fset *pflag.FlagSet) ([]string, error) {
+	clauses := []string{}
+
+	state := fset.Lookup("state")
+	if state.Changed {
+		val, ok := stateOptionMap[state.Value.String()]
+		if !ok {
+			return nil, fmt.Errorf("%w: %s", errInvalidStateFilter, state.Value.String())
+		}
+		clauses = append(clauses, fmt.Sprintf("state = %s", val.String()))
+	}
+
+	// We have a series of string options we need to map to the upstream API filters
+	for ours, theirs := range map[string]string{
+		"ip-address":                    "source.ip_address",
+		"client-id":                     "client_id",
+		"client-software-name":          "client_software_name",
+		"client-software-version":       "client_software_version",
+		"group-id":                      "group_id",
+		"authentication-user-principal": "authentication_info.user_principal",
+	} {
+		flag := fset.Lookup(ours)
+		if flag.Changed {
+			clauses = append(clauses, fmt.Sprintf("%s = %q", theirs, flag.Value.String()))
+		}
+	}
+
+	idleFlag := fset.Lookup("idle-ms")
+	if idleFlag.Changed {
+		idleMs, err := fset.GetInt64("idle-ms")
+		if err != nil {
+			return nil, err
+		}
+
+		clauses = append(clauses, fmt.Sprintf("idle_duration > %dms", idleMs))
+	}
+	return clauses, nil
+}
+
+func newConnectionList(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Display statistics about current connections",
+		Long:  `Display statistics about current connections.`,
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			prof, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			cl, err := adminapi.NewClient(cmd.Context(), fs, prof)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
```

Comment:
If this command uses the admin API and is not yet in the public API then it won't be usable for cloud clusters, you could add

```
config.CheckExitCloudAdmin(p)
```